pyproject.toml

```
@@ -61,7 +61,7 @@ tenacity = ">=8.2.3,<9.0.0"
 pyarrow = { version = ">=9.0.0,<18.0.0", optional = true }
 pandas = { version = ">=1.0.0,<3.0.0", optional = true }
 duckdb = { version = ">=0.5.0,<2.0.0", optional = true }
-ray = { version = ">=2.0.0,<2.10.0", optional = true }
+ray = { version = "==2.10.0", optional = true }
```

I think we can be a bit more permissive here:
```suggestion
ray = { version = ">=2.10.0,<3.0.0", optional = true }
```

---

I think I found the reason why the previous version didn't run - I think I can resolve it with python version based version overrides - will put up a fix shortly

---

This is the issue I'm running into with enabling higher version of ray:

```
  - Installing ray (2.32.0): Failed

  RuntimeError

  Unable to find installation candidates for ray (2.32.0)

  at /usr/local/python/3.10.13/lib/python3.10/site-packages/poetry/installation/chooser.py:74 in choose_for
       70│ 
       71│             links.append(link)
       72│ 
       73│         if not links:
    →  74│             raise RuntimeError(f"Unable to find installation candidates for {package}")
       75│ 
       76│         # Get the best link
       77│         chosen = max(links, key=lambda link: self._sort_key(package, link))
```

---

I've added in a python version based constraint so that only python3.8 has the restrictive constraint of 2.10.0. The higher ray versions don't have a binary published for python3.8: https://pypi.org/project/ray/2.11.0/#files